### ISSUE TITLE
Fix weird behavior with `ctrl+c`

### ISSUE THREAD

**Issue by waydegg** [biobootloader/mentat#82]
I seem to "randomly" have my `ctrl+c` keybind disabled during response streaming from OpenAI. `KeyboardInterrrupt` is getting swallowed somewhere, however I'm still trying to consistently reproduce this behavior.

**Comment by waydegg** [comment]
So this ended up not being an issue with anything in the main branch, so I'm closing this. 

I found this during development/debugging, but basically there's issues with throwing exceptions inside of a prompt toolkit keybind handler ([more context here](https://github.com/prompt-toolkit/python-prompt-toolkit/issues/1576)). The SIGINT/SIGTERM signals get "blocked" from traveling up the call stack due to how they're initialized. Prompt toolkit nests threads and async event loops, so getting the signals to be registered properly requires some changes w/ the library source code.

TL;DR don't throw exceptions inside of keybind handlers
